elide-integration-tests/src/test/java/example/AssignedIdLong.java

```diff
@@ -37,4 +37,20 @@ public int getValue() {
     public void setValue(int value) {
         this.value = value;
     }
+
+    @Override
+    public int hashCode() {
+        return Long.hashCode(id);
```

Comment:
The AssignedId classes are intended to test scenarios where Elide does not generate the ID but rather it is provided externally (by the client).  

In those scenarios, the ID is both the mechanism to identify uniqueness in the RDBMS AND uniqueness in Java.

BaseID is intended for scenarios where the ID is in flux (because the ORM may not have it if the database provides it), but we still need to a common mechanism to identify uniqueness.